Sources/NIO/Bootstrap.swift

```diff
@@ -967,3 +979,44 @@ public final class NIOPipeBootstrap {
         }
     }
 }
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandBootstrapOption {
+    private let apply : ShorthandApply
+    
+    private init(apply : ShorthandApply) {
+        self.apply = apply
+    }
+    
+    /// Apply the contained option to the supplied ServerBootstrap
+    /// - Parameter serverBootstrap: bootstrap to apply this option to.
+    /// - Returns:the modified bootstrap (currently the same one mutated)
+    func applyOption<BootstrapT : ServerBootstrap>(serverBootstrap : BootstrapT) -> BootstrapT {
```

Comment:
I don't think this function is _actually_ generic: `ServerBootstrap` is a concrete type, not a protocol, so nothing conforms to it. The generic can be removed and just take a concrete `ServerBootstrap`.

Reply:
Sadly because the calling function returns a Self, the compiler doesn't seem smart enough to realise that ServerBootstrap is the only possible answer.

Reply:
I don't think that's an issue: it should be totally fine to have the signature of this function be:

```
func applyOption(serverBootstrap: ServerBootstrap) -> ServerBootstrap
```

The compiler should be quite able to sort that out.

Reply:
Sadly, this is exactly how I started - I added the generics to get it compiling.  There is an alternative which I found which does work which I'll show you later.

Reply:
Changed how this was done to work.